File: backend/app/utils/statuses.py
"""Application status constants."""

APPLICATION_STATUSES = [
    "pending",
    "under_review",
    "approved",
    "rejected",
    "released",
    "partially_paid",
    "paid",
    "overdue",
    "defaulted",
    "cancelled",
]

APPLICATION_STATUS_PATTERN = "^(" + "|".join(APPLICATION_STATUSES) + ")$"

ALLOWED_STATUS_TRANSITIONS = {
    "pending": {"under_review"},
    "under_review": {"approved", "rejected"},
    "approved": {"released", "cancelled"},
    "released": {"partially_paid", "paid", "cancelled"},
    "partially_paid": {"paid"},
    "overdue": {"paid", "defaulted"},
    "defaulted": {"paid"},
    "paid": {"approved"},
}

File: frontend/src/utils/statuses.js
export const APPLICATION_STATUSES = [
  { value: 'pending', label: 'Pending' },
  { value: 'under_review', label: 'Under Review' },
  { value: 'approved', label: 'Approved' },
  { value: 'rejected', label: 'Rejected' },
  { value: 'released', label: 'Released' },
  { value: 'active', label: 'Active' },
  { value: 'partially_paid', label: 'Partially Paid' },
  { value: 'paid', label: 'Paid' },
  { value: 'overdue', label: 'Overdue' },
  { value: 'defaulted', label: 'Defaulted' },
  { value: 'cancelled', label: 'Cancelled' },
]

export const getStatusLabel = (status) => {
  return APPLICATION_STATUSES.find((item) => item.value === status)?.label || status
}
